### PR TITLE
Remove unneeded trailing slash of void elements

### PR DESCRIPTION
**Pull request by ann0see**
**Short description of changes**

HTML void elements should not use trailing slashes:  https://github.com/validator/validator/wiki/Markup-%C2%BB-Void-elements#trailing-slashes

This removes (hopefully) all of them
**Context: Fixes an issue? Related issues**
Makes w3c validator happy

**Status of this Pull Request**
Ready for review.

**What is missing until this pull request can be merged?**
Review

**Does this need translation?**

No.

## Checklist

- [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
- [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
- [x] I'm sure that this Pull Request goes to the correct branch

### CODE REVIEW

**Comment by ann0see** [comment]
Po4a doesn't seem to like removing "/"? Checks fail.

**Comment by pljones** [comment]
See https://www.thoughtco.com/why-different-html-versions-3471349

**Comment by ann0see** [comment]
I thought we're using HTML5?

**Comment by pljones** [comment]
> I thought we're using HTML5?

At a guess, do files declare that with a doctype?  If we're leaving parsers to guess, it's easy to guess wrongly.

**Comment by ann0see** [comment]
We do have a doctype html tag, so it should be interpreted as html5.

**Comment by pljones** [comment]
In `Update .po files` I can see:
```
../wiki/en/misc/1-index.html:48: (po4a::xml)
Unexpected closing tag </p> found. The main document may be wrong.
```
when updating the language files.  Does that mean po4a thinks it's using `xml` style tags?  That's also reported in `Create translated docs and stats`.  I'd check for an _extra_ `</p>` where one isn't needed based on the error report.

----

It looks like that then has a knock on effect as, later, in `Build the site in the jekyll/builder container`:
```
Installing jekyll-minifier 0.1.10
Bundle complete! 4 Gemfile dependencies, 37 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
ruby 3.1.1p18 (2022-02-18 revision 53f5fc4236) [x86_64-linux-musl]
Configuration file: /srv/jekyll/_config.yml
            Source: /srv/jekyll
       Destination: /srv/jekyll/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
/usr/gem/gems/jekyll-4.2.2/lib/jekyll/utils.rb:141:in `initialize': No such file or directory @ rb_sysopen - /srv/jekyll/1-nl-index.html (Errno::ENOENT/usr/gem/gems/jekyll-4.2.2/lib/jekyll/utils.rb:141:in `initialize': No such file or directory @ rb_sysopen - /srv/jekyll/1-nl-index.html (Errno::ENOENT)
	from /usr/gem/gems/jekyll-4.2.2/lib/jekyll/utils.rb:141:in `open'Installing jekyll-minifier 0.1.10
Bundle complete! 4 Gemfile dependencies, 37 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
...
```
and then fails.

Then, `Zip Website` reports
```
  zip -r /home/runner/work/jamuluswebsite/jamuluswebsite/website.zip _site/*
  shell: /usr/bin/bash -e {0}
	zip warning: name not matched: _site/*

zip error: Nothing to do! (try: zip -r /home/runner/work/jamuluswebsite/jamuluswebsite/website.zip . -i _site/*)
Error: Process completed with exit code 12.
```

**Comment by ann0see** [comment]
> Does that mean po4a thinks it's using xml style tags? 

Yes, that's what I suspect.

**Comment by pljones** [comment]
Mm.  Nothing obvious like unbalanced tags...  The [document itelf](https://github.com/jamulussoftware/jamuluswebsite/blob/release/wiki/en/misc/1-index.html) doesn't have a `<doctype>` as it's a fragment.  Does po4a have a config that's fed in?

**Comment by ann0see** [comment]
I searched for it but didn't find anything. 

@ignotus666 ?

**Comment by ignotus666** [comment]
I can't really look into it properly yet until Monday but yes, html docs are processed as xml by po4a. I think the checks for tag "errors" can be disabled or left at just a warning (without aborting the file processing), as it seems to default to aborting. There's info about this in the official po4a documentation. It should be a matter of adding a config parameter to the scripts.

**Comment by pljones** [comment]
I'd really like to leave the config until we identify what's changed to cause it to start breaking.

**Comment by ignotus666** [comment]
[Here's](https://po4a.org/man/man3/Locale::Po4a::Xml.3pm.php) the relevant page in the po4a docs - scroll down to "ontagerror". It seems that po4a interprets that `<br>` tag as an opening tag and doesn't like that the next closing tag doesn't match it. Turns out the update I was preparing for #817 would "fix" this by just printing a warning (and not exiting). I don't know if that's an acceptable solution though.

On closer inspection, I was using "ontagerror", but maybe "optionalclosingtag" would be a better choice?

**Comment by pljones** [comment]
Does HTML5 allow
```html
<p>paragraph one
<p>paragraph two without a previous closing &lt;/p&gt;</p>
```
?  Do we want to allow that?  It sounds like "ontagerror" would cause the error - but also errors on `<br>` rather than `<br/>` and "optionalclosingtag" would allow the "missing" `</p>` whilst allowing `<br>`.

**Comment by ignotus666** [comment]
Another option is to use the "Xhtml" instead of "xml" module in po4a. It correctly deals with those tags without adding options but segments the paragraphs in a completely different way, meaning translators would have to re-translate "1-Index.html", though it seems a re-write is in the works so maybe not such a big deal.

[Here's](https://github.com/mquinson/po4a/issues/179) an issue raised about the same problem that arose with the Xhtml module, but which seems to have been solved since.

File: _includes/general/navbtn.html [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <svg role="img" width="20" height="20" aria-labelledby="ntitle" id="navbtn" class="hide-on-dt">
   <title id="ntitle">{{ site.data.general.nav.btnOpenNavigation }}</title>
   <g>
-    <rect fill="currentcolor" id="r1" width="100%" height="3" x="0" y="0" />
-    <rect fill="currentcolor" id="r2" width="100%" height="3" x="0" y="8" />
-    <rect fill="currentcolor" id="r3" width="100%" height="3" x="0" y="16" />
+    <rect fill="currentcolor" id="r1" width="100%" height="3" x="0" y="0">
```
Comment (ann0see) [inline _includes/general/navbtn.html]: Firefox mobile dev mode doesn't seem to like this change. The last two hamburger rectangles disappear.  SVG's need self closing tags.

Comment (ann0see) [reply _includes/general/navbtn.html]: I think I fixed it now.

**Review by ann0see — COMMENTED** [review]
I think all is ok now after the update by ignotus.

**Comment by ignotus666** [comment]
There's something wrong - all the file updates with po4a fail, but the script doesn't exit and the check looks like it passed when it didn't. Apparently`libs-parse-keyword-perl` needs to be installed. I'm going to add it to the cache script, and change the key back to "po4a" to force a new cache to be created. Then re-run this action to see if it's fixed.

**Comment by ignotus666** [comment]
Ok, fixed now.

**Review by ignotus666 — APPROVED** [review]

**Comment by ignotus666** [comment]
Aargh, this has caused Weblate to lock due to a whole load of conflicts...

**Pull request merged**